### DISCUSSION TITLE
Translation

### DISCUSSION THREAD

**Discussion by DavidMacha** [openai/openai-python#755] [Q&A] [answered]
Hi, everyone! I'm facing a problem with the openai-python library. I've integrated it into my language translation application, and I'm using GPT-3 to translate English text to French. The translations I'm getting are not as accurate and coherent as I expected. Can someone help me troubleshoot this issue?

**Comment by Mattral** [comment]
Can you share a code snippet where you're making the API call to GPT-3 for translation? It will be easier to pinpoint the problem with the specific code.

**Reply by DavidMacha** [reply]
```
import openai

api_key = "my_api_key_here"
openai.api_key = api_key

prompt = "Translate the following English text to French: 'Hello, how are you?'"

response = openai.Completion.create(
    engine="text-davinci-002",
    prompt=prompt,
    max_tokens=50  # I tried adjusting the token count, but it didn't help much
)

print(response.choices[0].text)

```

**Reply by Mattral** [reply]
The code looks fine. However, one thing to consider is that GPT-3's output can sometimes be sensitive to the wording of the prompt. It's crucial to make your instruction more explicit. Instead of asking it to "Translate," you might try something like, "Translate the English sentence, 'Hello, how are you?' into French." This way, GPT-3 has a clearer instruction.

**Reply by Mattral** [reply, marked as answer]
Another thing to consider, is the choice of the GPT-3 engine. Depending on the nature of your task, some engines might perform better. You might want to experiment with different engines and see if one provides more accurate translations for your specific use case.

**Reply by DavidMacha** [reply]
That's a great point! I'll experiment with different engines and take into account all the advice. Thanks for your help and suggestions. I'll work on improving the translation quality in my application.
